File: client/DabaebMinton/Assets/Scripts/Network/ReplicationHeader.cs
using System;
using UnityEngine;

using static NetworkUtils;

public class ReplicationHeader
{
    public ReplicationAction Ra { get; private set; }
    public uint Nid { get; private set; }

    public void Read(InputMemoryBitStream inStream)
    {
        Nid = (uint)inStream.ReadInt();
        Ra = (ReplicationAction)inStream.ReadInt(GetRequiredBits(ReplicationAction.RA_Max));
    }
};

File: server/Src/DabaebmintonServerCore/ReplicationHeader.cpp
#include "ServerPCH.h"
#include "ReplicationHeader.h"

#include "OutputMemoryBitStream.h"
#include "GetRequiredBits.h"

void ReplicationHeader::Write(OutputMemoryBitStream& inStream)
{
	inStream.WriteBits(&_ra, GetRequiredBits(static_cast<int>(ReplicationAction::RA_Max)));
	inStream.Write(_nid);
	inStream.Write(_cid);
}
